File: BeetleX.Blog/Controller/AdminBlog.js
/************************************************************************************
FastHttpApi javascript api Generator Copyright © henryfan 2018 email:henryfan@msn.com
https://github.com/IKende/FastHttpApi
**************************************************************************************/




var AdminBlogModifyUrl='/admin/blog/Modify';
/**
* 'var result= await AdminBlogModify(params);'
**/
function AdminBlogModify(id,title,top,tag,category,sourceUrl,content,summary,useHttp)
{
    return api(AdminBlogModifyUrl,{id:id,title:title,top:top,tag:tag,category:category,sourceUrl:sourceUrl,content:content,summary:summary},useHttp,true).sync();
}
/**
* 'AdminBlogModifyAsync(params).execute(function(result){},useHttp);'
**/
function AdminBlogModifyAsync(id,title,top,tag,category,sourceUrl,content,summary,useHttp)
{
    return api(AdminBlogModifyUrl,{id:id,title:title,top:top,tag:tag,category:category,sourceUrl:sourceUrl,content:content,summary:summary},useHttp,true);
}
var AdminBlogAllSyncToESUrl='/admin/blog/AllSyncToES';
/**
* 'var result= await AdminBlogAllSyncToES(params);'
**/
function AdminBlogAllSyncToES(useHttp)
{
    return api(AdminBlogAllSyncToESUrl,{},useHttp).sync();
}
/**
* 'AdminBlogAllSyncToESAsync(params).execute(function(result){},useHttp);'
**/
function AdminBlogAllSyncToESAsync(useHttp)
{
    return api(AdminBlogAllSyncToESUrl,{},useHttp);
}
var AdminBlogDeleteUrl='/admin/blog/Delete';
/**
* 'var result= await AdminBlogDelete(params);'
**/
function AdminBlogDelete(id,useHttp)
{
    return api(AdminBlogDeleteUrl,{id:id},useHttp,true).sync();
}
/**
* 'AdminBlogDeleteAsync(params).execute(function(result){},useHttp);'
**/
function AdminBlogDeleteAsync(id,useHttp)
{
    return api(AdminBlogDeleteUrl,{id:id},useHttp,true);
}
var AdminBlogGetUrl='/admin/blog/Get';
/**
* 'var result= await AdminBlogGet(params);'
**/
function AdminBlogGet(id,useHttp)
{
    return api(AdminBlogGetUrl,{id:id},useHttp).sync();
}
/**
* 'AdminBlogGetAsync(params).execute(function(result){},useHttp);'
**/
function AdminBlogGetAsync(id,useHttp)
{
    return api(AdminBlogGetUrl,{id:id},useHttp);
}
var AdminBlogListUrl='/admin/blog/List';
/**
* 'var result= await AdminBlogList(params);'
**/
function AdminBlogList(category,index,useHttp)
{
    return api(AdminBlogListUrl,{category:category,index:index},useHttp).sync();
}
/**
* 'AdminBlogListAsync(params).execute(function(result){},useHttp);'
**/
function AdminBlogListAsync(category,index,useHttp)
{
    return api(AdminBlogListUrl,{category:category,index:index},useHttp);
}

File: BeetleX.Blog/Controller/AdminBlog.cs
using System;
using System.Collections.Generic;
using System.Text;
using BeetleX.FastHttpApi;
using Peanut;
using BeetleX.Blog.DBModules;
using System.Linq;
namespace BeetleX.Blog.Controller
{
    [Controller(BaseUrl = "/admin/blog")]
    [AdminFilter]
    public class AdminBlog
    {
        [Post]
        public void Modify(long id, string title, bool top, string tag, long category, string sourceUrl, string content,
            string summary)
        {
            DBModules.Blog item;
            if (id > 0)
            {
                item = DBContext.Load<DBModules.Blog>(id);
            }
            else
            {
                item = new DBModules.Blog();
            }
            if (item != null)
            {
                item.Title = title;
                item.Top = top;
                item.Tags = tag;
                item.CategoryID = category;
                Category cate = DBContext.Load<Category>(category);
                if (cate != null)
                    item.Category = cate.Name;
                item.SourceUrl = sourceUrl;
                item.Content = content;
                item.Summary = summary;
                item.Save();
                RefreshBlog refreshBlog = new RefreshBlog();
                refreshBlog.BlogID = item.ID;
                refreshBlog.Status = 1;
                refreshBlog.Save();
            }
        }

        public void AllSyncToES()
        {
            foreach (DBModules.Blog item in new Expression().List<DBModules.Blog>())
            {
                RefreshBlog refreshBlog = new RefreshBlog();
                refreshBlog.BlogID = item.ID;
                refreshBlog.Status = 1;
                refreshBlog.Save();
            }
        }

        [Post]
        public void Delete(long[] id)
        {
            if (id != null && id.Length > 0)
            {
                (DBModules.Blog.iD == id).Delete<DBModules.Blog>();
                (DBModules.Comment.blogID == id).Delete<DBModules.Comment>();
                foreach (var item in id)
                {
                    RefreshBlog refreshBlog = new RefreshBlog();
                    refreshBlog.BlogID = item;
                    refreshBlog.Status = 0;
                    refreshBlog.Save();
                }
            }
        }

        public object Get(long id)
        {
            var item = DBContext.Load<DBModules.Blog>(id);
            if (item != null)
                return new { item.ID, item.Title, item.Content, item.CategoryID, item.SourceUrl, item.Tags, item.Top };
            return new object();
        }

        public Object List(long category, int index)
        {
            int size = 15;
            Expression exp = new Expression();
            if (category > 0)
                exp &= DBModules.Blog.categoryID == category;
            int count = exp.Count<DBModules.Blog>();
            int pages = count / size;
            if (count % size > 0)
                pages++;
            var datas = exp.List<DBModules.Blog>(new Region(index, size), DBModules.Blog.createTime.Desc);
            var items = from a in datas select new { a.ID, a.Title, a.Category, a.Top, CreateTime = a.CreateTime.ToString() };
            return new { Index = index, Pages = pages, Items = items };
        }
    }
}
